rzv2h-evk: Support ADC and enable some kernel configuation

long-trinh-rvc opened the pull request:
Add ADC driver to RZ/V2H and handle some issue on Audacity on Ubuntu LXDE

Commit:
rzv2h-evk: tty: serial: rz-sci: Add Renesas SCI driver

Add Renesas SCI driver for RZ/G3E.
It is based on sh_sci.c.

Signed-off-by: Linh Hua <linh.hua.jc@renesas.com>
Signed-off-by: Nghia Vo <nghia.vo.zn@renesas.com>
Signed-off-by: Long Trinh <long.trinh.rh@bp.renesas.com>

Commit:
driver: iio: adc: Add ADC driver for RZ/V2H

This patch add Analogue to Digital Converter (ADC) driver for RZ/V2H SoC.

Signed-off-by: Huy Nguyen <huy.nguyen.wh@renesas.com>
Signed-off-by: Nghia Vo <nghia.vo.zn@renesas.com>

Commit:
rzv2h-evk: include: mfd: rz-mtu3: add some missing RZ/MTU3's definition

Add some missing RZ/MTU3's definition:
- Enable output setting for TOER, TSTR and TDER register
- Define offset address some registers such as: TOERA, TOERB, TDERA, TDERB

Signed-off-by: Tranh Ha <tranh.ha.xb@renesas.com>
Signed-off-by: Long Trinh <long.trinh.rh@bp.renesas.com>

Commit:
rzv2h-evk: include: mfd: rz-mtu3: add some missing RZ/MTU3's definition

Add some missing RZ/MTU3's definition:
- Enable output setting for TOER, TSTR and TDER register
- Define offset address some registers such as: TOERA, TOERB, TDERA, TDERB

Signed-off-by: Tranh Ha <tranh.ha.xb@renesas.com>
Signed-off-by: Long Trinh <long.trinh.rh@bp.renesas.com>

Commit:
drivers: mfd: rz-mtu3: Modified driver and include to support feature clock source, clock event

This commit modified struct in include and driver rz-mtu3.c to support
feature clock source, clock event

Signed-off-by: Tu Nguyen <tu.nguyen.xg@renesas.com>

hienhuynh2809 committed:
rzv2h-evk: clocksource: rz_mtu3_clk: Add timer usage support for MTU3

Add timer usage support for MTU3 which is available in RZ/G2L Series,
RZ/V2L and RZ/Five.

Signed-off-by: Hien Huynh <hien.huynh.px@renesas.com>
Signed-off-by: Long Trinh <long.trinh.rh@bp.renesas.com>

hienhuynh2809 committed:
rzv2h-evk: clocksource: rz_mtu3_clk: Add timer usage support for MTU3

Add timer usage support for MTU3 which is available in RZ/G2L Series,
RZ/V2L and RZ/Five.

Signed-off-by: Hien Huynh <hien.huynh.px@renesas.com>
Signed-off-by: Long Trinh <long.trinh.rh@bp.renesas.com>

Commit:
tty: serial: rz-sci: Add missing get reset for RSCI driver

Add support get more than one reset in driver since it has
been removed in TFA.

Signed-off-by: Nghia Vo <nghia.vo.zn@renesas.com>

Commit:
rzv2h-evk: Add i3c device userspace interface

Currently there is no way to use i3c devices from user space,
this patch will support userspace interface for I3C.

Refer source:
https://patchwork.kernel.org/project/linux-i3c/cover/cover.1575977795.git.vitor.soares@synopsys.com/

Signed-off-by: Vitor Soares <Vitor.Soares@synopsys.com>

Signed-off-by: Au Dang <au.dang.jx@renesas.com>
Signed-off-by: Long Trinh <long.trinh.rh@bp.renesas.com>

Commit:
serial: rz-sci: Fix infinite framing error warnings.

RXDESEL: Select start receiving condition.
- If this bit is 0, receiving process will start when RX line at LOW,
this can lead to infinite framing error warnings.
- If this bit is 1, the receiving process will start
when FALLING EDGE detected. This option is recommended.

Signed-off-by: Linh Hua <linh.hua.jc@renesas.com>
Signed-off-by: Nghia Vo <nghia.vo.zn@renesas.com>

hienhuynh2809 committed:
tty: serial: rz-sci: correct setting for kernel console writing

Correct setting for kernel console writing:
- Remove CCR0_TIE (TX empty interrupt enable) for polling method
to avoid conflict with userspace console.
- Based on HW manual, CCR0_TIE and CCR0_TE must be set to 1
simultaneously to trigger TX empty interrupt. So, there are some
cases kernel space console can be called before TX end interrupt
that can make CCR0_TIE change from 0 to 1 but CCR0_TE remains 1.
- Should clear CCR0_TE bit before restoring CCR0 register in kernelspace
console.

Signed-off-by: hienhuynh <hien.huynh.px@renesas.com>

hienhuynh2809 committed:
tty: serial: rz-sci: avoid TE is cleared when user TX data is still remained

In low latency environment, TX empty or sci_start_tx can preempt
TX end interrupt handler to make the wrong procedure that CCR0_TE
is cleared when TX in transmitting.
So, disable TXI (empty) after sending the last data to ensure
TX end irq can not be preempted by TXI irq.
Also, re-enable TE and TIE bits before finishing TX end
if there is still pending data in uart buffer to let them flush
the remained data.

Signed-off-by: hienhuynh <hien.huynh.px@renesas.com>

Commit:
serial: rz-sci: Using dynamically request major number for rsci

Replace static major number to dynamic major number to prevent
static major number is registered by other module.

Signed-off-by: Tu Nguyen <tu.nguyen.xg@renesas.com>

Commit:
tty: serial: rz-sci: Add reset_control_{assert,deassert} for S2RAM

Adds reset_control_{assert,deassert} in PM_SLEEP handlers
as a requirement when Suspend-to-ram in RZ/G3S, RZ/G3E,
RZ/G2L platforms.

Signed-off-by: Nghia Nguyen <nghia.nguyen.xm@renesas.com>
Signed-off-by: Tu Nguyen <tu.nguyen.xg@renesas.com>

Commit:
iio: adc: rzv2h_adc: Add PM_SLEEP suspend/resume support

Adds S2RAM suspend/resume handlers to properly
backup/restore ADC controller settings for S2RAM.

Signed-off-by: Thang Dinh <thang.dinh.fv@renesas.com>

Commit:
driver: iio: adc: Support mstp_ada active ADC

This patch supports hanler function for mstp_ada bit by using sysc
driver.

Signed-off-by: Thang Dinh <thang.dinh.fv@renesas.com>

Commit:
clocksource: rz_mtu3_clk: Rename pm_genpd_syscore_poweron|off

pm_genpd_syscore_poweroff and pm_genpd_syscore_poweron
has been renamed after this commit
fc51989062138744b56e47190915ce68484e73f3
using dev_pm_genpd_suspend|resume for replace.

Remove 2 callings of clk_disable in rz_mtu3_clk_setup function.
Remove calling of clk_enable in rz_mtu3_clk_enable function.
Remove calling of clk_disable in rz_mtu3_clk_disable function in
order to keep balance between clk_enable and clk_disable in driver.

Signed-off-by: Thang Dinh <thang.dinh.fv@renesas.com>

Commit:
rzv2h-evk: clocksource: rz_mtu3_clk: Rename pm_genpd_syscore_poweron|off

pm_genpd_syscore_poweroff and pm_genpd_syscore_poweron
has been renamed after this commit
fc51989062138744b56e47190915ce68484e73f3
using dev_pm_genpd_suspend|resume for replace.

Remove 2 callings of clk_disable in rz_mtu3_clk_setup function.
Remove calling of clk_enable in rz_mtu3_clk_enable function.
Remove calling of clk_disable in rz_mtu3_clk_disable function in
order to keep balance between clk_enable and clk_disable in driver.

Signed-off-by: Thang Dinh <thang.dinh.fv@renesas.com>
Signed-off-by: Long Trinh <long.trinh.rh@bp.renesas.com>

Commit:
driver: iio: adc: Update adc sysc signal

This patch add setting bit adc signal when suspend in oder to
keep balance with adc signal when resume.

Signed-off-by: Thang Dinh <thang.dinh.fv@renesas.com>

long-trinh-rvc committed:
rzv2h-evk: adc: add config to renesas_defconfig

This commit adds default adc config to renesas_defconfig

Signed-off-by: Long Trinh <long.trinh.rh@bp.renesas.com>

long-trinh-rvc committed:
arm64: dts: renesas: add sysc argument to ADC node

This commit allows the system controller to take control of the ADC.

Signed-off-by: Long Trinh <long.trinh.rh@bp.renesas.com>

tiennguyenuh commented on sound/soc/codecs/da7213.c:
```diff
@@ -2117,6 +2117,9 @@ static int da7213_probe(struct snd_soc_component *component)
 	/* Default infinite tone gen, start/stop by Kcontrol */
 	snd_soc_component_write(component, DA7213_TONE_GEN_CYCLES, DA7213_BEEP_CYCLES_MASK);
 
+	/* Default infinite tone gen, start/stop by Kcontrol */
+	snd_soc_component_write(component, DA7213_TONE_GEN_CYCLES, DA7213_BEEP_CYCLES_MASK);
+
```
Is that redundant? These two lines above look identical

long-trinh-rvc replied:
Yes, I will remove it

vudangRVC commented on sound/soc/codecs/da7213.c:
```diff
@@ -1559,7 +1559,7 @@ static int da7213_set_component_sysclk(struct snd_soc_component *component,
 		return 0;
 
-	if (((freq < 5000000) && (freq != 32768)) || (freq > 54000000)) {
+	if (((freq < 5000000) && (freq != 32768) && (freq != 2822400)) || (freq > 54000000)) {
```
Can you provide the background of this change?

long-trinh-rvc replied:
When starting Audacity, it shows some error logs indicating that DA7213 does not support the MCLK value of 2,822,400, leading to hardware failure
```
[   99.660839] da7213 2-001a: Unsupported MCLK value 2822400
[   99.666322] da7213 2-001a: ASoC: error at snd_soc_component_set_sysclk on da7213.2-001a: -22
[   99.674821]  rsnd-dai.0-da7213-hifi: ASoC: error at snd_soc_link_hw_params on rsnd-dai.0-da7213-hifi: -22
[   99.684567] rcar_sound 13c00000.sound: audmac-pp[0] : hw_free error (0, -1)
[   99.691510] rcar_sound 13c00000.sound: audmac[1] : hw_free error (0, -1)
[   99.698188] rcar_sound 13c00000.sound: dvc[0] : hw_free error (0, -1)
[   99.704605] rcar_sound 13c00000.sound: cmd[0] : hw_free error (0, -1)
[   99.711023] rcar_sound 13c00000.sound: src[0] : hw_free error (0, -1)
[   99.717442] rcar_sound 13c00000.sound: ssiu[40] : hw_free error (0, -1)
[   99.724032] rcar_sound 13c00000.sound: ssi[3] : hw_free error (0, -1)
[   99.730449] rcar_sound 13c00000.sound: ssi[4] : hw_free error (0, -1)
[   99.736866] rcar_sound 13c00000.sound: audmac-pp[0] : hw_free error (0, -1)
```
It actually does not affect the recording ability in Audacity, so I intend to remove this change in the next commit.

long-trinh-rvc committed:
rzv2h-evk: rz-sci: remove some redundant code

This commit removes some redundant code that is no longer usable on 6.10.

Signed-off-by: Long Trinh <long.trinh.rh@bp.renesas.com>

long-trinh-rvc committed:
rzv2h-evk: renesas_defconfig: add SPI_SLAVE configuration

This commit adds CONFIG_SPI_SLAVE and its related configuation:
    - CONFIG_SPI_SLAVE_TIME
    - CONFIG_SPI_SLAVE_SYSTEM_CONTROL

Signed-off-by: Long Trinh <long.trinh.rh@bp.renesas.com>
